build-mbl/poky-build/Dockerfile

```diff
@@ -0,0 +1,159 @@
+FROM ubuntu:18.04
+
+# if running as a jenkins slave:
+# 1. docker build --tag jenkins/ww-build-env:latest --build-arg mode=jenkins .
```

Comment:
changed to use a Dockerfile based upon the mbl version, which is simpler. The build scripts that are copied are different between the two.

Reply:
Thanks. So now, the diff between the usual MBL dockerfile and the poky-test dockerfile is:
```Diff
--- Dockerfile	2019-06-27 14:39:50.188262900 +0100
+++ poky-test/Dockerfile	2019-07-15 09:29:07.611525704 +0100
@@ -39,6 +39,7 @@ RUN apt-get update && apt-get install -y
     xz-utils
 
 RUN apt-get update && apt-get install gosu
+RUN pip3 install --upgrade pip
 RUN pip3 install dohq-artifactory && pip3 install jinja2
 COPY entrypoint.sh /usr/local/bin/entrypoint.sh
 
@@ -62,3 +63,7 @@ COPY git-setup.sh ssh-setup.sh build.sh
      README.binary_release_template \
      license-report \
      Licensing_and_Acknowledgment.template_nxp ./
+
+# Scripts used to build poky-psa image
+COPY poky-test/build-poky.py support/file_util.py poky-test/poky-bitbake-wrapper.sh git-setup.sh ssh-setup.sh ./
+
```

So:
* Why is the pip upgrade required for poky-test but not MBL? Should we also have it for MBL?
* In the added `COPY` command, I don't think you need to add `git-setup.sh` and `ssh-setup.sh` - these are added in the previously existing `COPY` command above... Could we just add these extra files to the MBL dockerfile too and use the same dockerfile for both types of build?


As an aside: are these pip commands bad for our build reproducibility? I sort-of assume that the apt-get commands won't have the same problem because I sort-of assume that the base ubuntu:xenial-20170517.1 image has configured apt sources to be static package repos... Maybe that's not true.

Reply:
pip update removed as not required on Jenkins.
The two dockerfiles are now almost identical except copying the specific build scripts. I could not find a way of having conditional COPY commands in the dockerfile. It could be done in the same dockerfile and copy the build scripts for the two current targets but I've left them separate as an example.

Reply:
> I could not find a way of having conditional COPY commands in the dockerfile.

I'd probably just copy all of the scripts all of the time - it should be harmless to have a couple of extra scripts hanging around in the MBL build.

Anyway, if the two dockerfiles are the same except for a bit extra at the end then you should use `FROM` to remove the duplication.

Reply:
As Pete and I discussed IRL: you can't use `FROM` to extend another dockerfile, you have to extend an _image_. So you'd have to build the MBL image before building the poky-test image.

Reply:
Combined both Dockerfiles into a single mbl/Dockerfile that will be used for mbl and poky builds.

Reply:
Thanks. Resolving.